systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/KafkaAdminClient.java

```diff
@@ -28,101 +29,64 @@
-        partitions = (builder.partitions <= 0) ? 1 : builder.partitions;
-        topicCount = (builder.topicCount <= 0) ? 1 : builder.topicCount;
-        topicOffset = builder.topicOffset;
-        adminName = builder.adminName;
-        bootstrapAddress = builder.bootstrapAddress;
-        topicName = builder.topicName;
-        additionalConfig = builder.additionalConfig;
-        namespaceName = builder.namespaceName;
-        topicOperation = builder.topicOperation;
-    }
-
-    public String getAdminName() {
-        return adminName;
-    }
+    public KafkaAdminClient(KafkaAdminClientsBuilder builder) {
+        super(builder);
+        checkParametersForBuilder(builder);
 
-    public String getBootstrapAddress() {
-        return bootstrapAddress;
+        replicationFactor = (builder.getReplicationFactor() <= 0) ? 1 : builder.getReplicationFactor();
+        partitions = (builder.getPartitions() <= 0) ? 1 : builder.getPartitions();
+        topicCount = (builder.getTopicCount() <= 0) ? 1 : builder.getTopicCount();
+        topicOffset = builder.getTopicOffset();
+        adminName = builder.getAdminName();
+        topicOperation = builder.getTopicOperation();
     }
 
-    public String getTopicName() {
-        return topicName;
+    @Override
+    protected KafkaAdminClientsBuilder newBuilder() {
+        return new KafkaAdminClientsBuilder();
     }
 
-    public int getTopicCount() {
-        return topicCount;
-    }
+    protected KafkaAdminClientsBuilder updateBuilder(KafkaAdminClientsBuilder builder) {
+        super.updateBuilder(builder);
 
-    public int getTopicOffset() {
-        return topicOffset;
-    }
-
-    public String getAdditionalConfig() {
-        return additionalConfig;
-    }
-
-    public String getNamespaceName() {
-        return namespaceName;
-    }
-
-    public int getPartitions() {
-        return partitions;
+        return builder
+            .withTopicOperation(topicOperation)
+            .withTopicCount(topicCount)
+            .withTopicOffset(topicOffset)
+            .withPartitions(partitions)
+            .withReplicationFactor(replicationFactor)
+            .withKafkaAdminClientName(adminName);
     }
 
-    public int getReplicationFactor() {
-        return replicationFactor;
+    @Override
+    public KafkaAdminClientsBuilder toBuilder() {
+        return updateBuilder(newBuilder());
     }
 
-    public String getTopicOperation() {
-        return topicOperation;
-    }
+    private void checkParametersForBuilder(KafkaAdminClientsBuilder builder) {
+        if (builder.getTopicOperation() == null || builder.getTopicOperation().isEmpty()) {
+            throw new InvalidParameterException("TopicOperation must be set.");
+        }
 
-    public KafkaAdminClient.Builder newBuilder() {
-        return new KafkaAdminClient.Builder();
-    }
+        if (builder.getBootstrapAddress() == null || builder.getBootstrapAddress().isEmpty()) {
```

Comment:
This is not needed I think. :) When you invoke `super(builder);` constructor it will call also the method from `BaseClients. checkParametersForBuilder`

Reply:
completely forgot about this :) 👍🤲